badge_server/main.py

```diff
@@ -76,518 +58,196 @@ def _get_pair_status_for_packages(pkg_sets):
-        # If not pre stored in BigQuery, run the check for the package.
-        else:
-            py2_res = badge_utils.checker.check([package_name], '2')
-            py3_res = badge_utils.checker.check([package_name], '3')
-
-            version_and_res['py2']['status'] = py2_res.get('result')
-            py2_description = py2_res.get('description')
-            py2_details = badge_utils.EMPTY_DETAILS if py2_description \
-                is None else py2_description
-            version_and_res['py2']['details'] = py2_details
-            version_and_res['py3']['status'] = py3_res.get('result')
-            py3_description = py3_res.get('description')
-            py3_details = badge_utils.EMPTY_DETAILS if py3_description \
-                is None else py3_description
-            version_and_res['py3']['details'] = py3_details
-
-        # Add the timestamp
-        version_and_res['timestamp'] = datetime.datetime.now().strftime(
-            badge_utils.TIMESTAMP_FORMAT)
-
-        # Write the result to Cloud Datastore
-        cache.set('{}_self_comp_badge'.format(package_key), version_and_res)
-
-    if not utils._is_package_in_whitelist([package_name]):
-        self_comp_res = badge_utils._build_default_result(
-            badge_type=badge_utils.BadgeType.SELF_COMP_BADGE,
-            status='UNKNOWN',
-            details=badge_utils.PACKAGE_NOT_SUPPORTED)
-    else:
-        self_comp_res = cache.get('{}_self_comp_badge'.format(package_key))
-
-    if self_comp_res is None:
-        details = version_and_res
-    else:
-        details = self_comp_res
-
-    # Run the check if details is None or forced to populate the cache or
-    # the cache is outdated.
-    if self_comp_res is None or force_run_check is not None:
-        threading.Thread(target=run_check).start()
-    elif self_comp_res is not None:
-        timestamp = self_comp_res.get('timestamp')
-        if not badge_utils._is_github_cache_valid(timestamp):
-            threading.Thread(target=run_check).start()
-
-    badge = badge_utils._get_badge(details, badge_name)
-    response = flask.make_response(badge)
-    response.content_type = badge_utils.SVG_CONTENT_TYPE
-    response.headers['Cache-Control'] = 'no-cache'
-    response.add_etag()
+    return (status, timestamp, self_compat_res, google_compat_res,
+            dependency_res)
 
-    return response
 
+@app.route('/')
+def greetings():
+    """This allows for testing server health using minimal resources"""
+    return 'Hello World!'
 
-@app.route('/self_compatibility_badge_target')
-def self_compatibility_badge_target():
-    """Return the dict which contains the self compatibility status and details
-    for py2 and py3.
 
-    e.g. {
-          'py2':{
-              'status': 'SUCCESS',
-              'details': None,
-          },
-          'py3':{
-              'status': 'CHECK_WARNING',
-              'details': '...',
-          }
-      }
+def _format_badge_name(pkgname, bdgname, commit_number):
```

Comment:
Use Google style for the doc string.

I don't think that this matches the PRD semantics.

I think that the logic is:

```python
if bdgname:
  return bdgname

if 'github.com' in pkgname:
  return 'compatibility check (master)'
else:
  return 'compatibility check (pypi)'
```
It might not be worth making a new function for this though.

Reply:
for the pypi case if no badge name is given, wouldn't it make sense to use the package name instead?

```
if bdgname:
  return bdgname

if 'github.com' in pkgname:
  return 'compatibility check (master)'
else:
  return '{} (pypi)'.format(pkgname)
```

Reply:
I concede and have applied the suggested change.